handlers/handler-loadbalance/src/main/java/io/servicecomb/loadbalance/filter/ZoneAwareServerListFilterExt.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Huawei Technologies Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.loadbalance.filter;
+
+import com.netflix.loadbalancer.Server;
+import io.servicecomb.loadbalance.CseServer;
+import io.servicecomb.loadbalance.ServerListFilterExt;
+import io.servicecomb.serviceregistry.RegistryUtils;
+import io.servicecomb.serviceregistry.api.registry.MicroserviceInstance;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ZoneAwareServerListFilterExt implements ServerListFilterExt {
+
+  @Override
+  public List<Server> getFilteredListOfServers(List<Server> list) {
+    List<Server> result = new ArrayList<>();
```

Comment:
is it possible to create cache for this?
avoid loop and build the list each time?